tests/fixtures/marine/marine_fixtures.py

```diff
@@ -25,6 +26,27 @@ def tcp_packet() -> bytes:
 
 
 @pytest.fixture(scope="session")
-def marine_instance() -> Marine:
-    path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "libmarine.so")
-    return Marine(path)
+def marine_so_path() -> str:
+    path = Path(__file__).parent / "libmarine.so"
+    return str(path.resolve())
+
+
+@pytest.fixture(scope="session")
+def marine_instance(marine_so_path):
+    return Marine(marine_so_path)
+
+
+@pytest.fixture(scope="session")
+def marine_pool_instance(marine_so_path):
+    return MarinePool(marine_so_path)
+
+
+@pytest.fixture(scope="session", params=[False, True], ids=["marine", "marine_pool"])
```

Comment:
Wouldn't it be more straight forward to pass the fixture names as params and use `request.getfixturevalue`?

Reply:
I don't think so. You add a line and a parameter (request) for each test

Reply:
I meant in here, not in every test